internal/dag/status.go

```diff
@@ -94,6 +94,10 @@ func (osw *ObjectStatusWriter) SetInvalid(desc string) {
 	osw.WithValue("description", desc).WithValue("status", StatusInvalid)
 }
 
+func (osw *ObjectStatusWriter) IsValid() bool {
```

Comment:
I'd prefer not to do this at this time. IsValid is a temporary state based on the primitive way we record status -- in the future it may become harder to determine if something is valid or not unless you know which path is being used to include it.

Reply:
Yeh, it's ugly. I'll look into the propagating errors to remove this.